File: src/main/java/com/aleprimo/Booking_System_App/service/OfferingService.java
package com.aleprimo.Booking_System_App.service;


import com.aleprimo.Booking_System_App.entity.Offering;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface OfferingService {
    Offering createOffering(Offering offering);
    Offering updateOffering(Long id, Offering offering);
    void deleteOffering(Long id);
    Optional<Offering> getOfferingById(Long id);
    Page<Offering> getAllOfferings(Pageable pageable);
    Page<Offering> getOfferingsByProviderId(Long providerId, Pageable pageable);
}

File: booking-frontend/src/services/offeringService.js
import api from "./api";

export const getOfferings = (page = 0, size = 10) =>
  api.get(`/offerings?page=${page}&size=${size}`);

export const getOfferingById = (id) => api.get(`/offerings/${id}`);

export const createOffering = (data) => api.post("/offerings", data);

export const updateOffering = (id, data) => api.put(`/offerings/${id}`, data);

export const deleteOffering = (id) => api.delete(`/offerings/${id}`);
